### PR TITLE
feat: Add option to include file path for Parquet, IPC, CSV scans

### PR DESCRIPTION
**Pull request by nameexhaustion**
Closes https://github.com/pola-rs/polars/issues/10481

This introduces an `include_file_paths` parameter, which accepts a column name to use for the column storing the file path.

```python
pl.scan_csv(
    ["https://j.mp/iriscsv", ".env/iris.csv"], include_file_paths="path"
).collect()
shape: (300, 6)
┌──────────────┬─────────────┬──────────────┬─────────────┬───────────┬─────────────────────────────────┐
│ sepal_length ┆ sepal_width ┆ petal_length ┆ petal_width ┆ species   ┆ path                            │
│ ---          ┆ ---         ┆ ---          ┆ ---         ┆ ---       ┆ ---                             │
│ f64          ┆ f64         ┆ f64          ┆ f64         ┆ str       ┆ str                             │
╞══════════════╪═════════════╪══════════════╪═════════════╪═══════════╪═════════════════════════════════╡
│ 5.1          ┆ 3.5         ┆ 1.4          ┆ 0.2         ┆ setosa    ┆ https://j.mp/iriscsv            │
│ 4.9          ┆ 3.0         ┆ 1.4          ┆ 0.2         ┆ setosa    ┆ https://j.mp/iriscsv            │
│ 4.7          ┆ 3.2         ┆ 1.3          ┆ 0.2         ┆ setosa    ┆ https://j.mp/iriscsv            │
│ 4.6          ┆ 3.1         ┆ 1.5          ┆ 0.2         ┆ setosa    ┆ https://j.mp/iriscsv            │
│ 5.0          ┆ 3.6         ┆ 1.4          ┆ 0.2         ┆ setosa    ┆ https://j.mp/iriscsv            │
│ …            ┆ …           ┆ …            ┆ …           ┆ …         ┆ …                               │
│ 6.7          ┆ 3.0         ┆ 5.2          ┆ 2.3         ┆ virginica ┆ file:////Users/nxs/git/polars/… │
│ 6.3          ┆ 2.5         ┆ 5.0          ┆ 1.9         ┆ virginica ┆ file:////Users/nxs/git/polars/… │
│ 6.5          ┆ 3.0         ┆ 5.2          ┆ 2.0         ┆ virginica ┆ file:////Users/nxs/git/polars/… │
│ 6.2          ┆ 3.4         ┆ 5.4          ┆ 2.3         ┆ virginica ┆ file:////Users/nxs/git/polars/… │
│ 5.9          ┆ 3.0         ┆ 5.1          ┆ 1.8         ┆ virginica ┆ file:////Users/nxs/git/polars/… │
└──────────────┴─────────────┴──────────────┴─────────────┴───────────┴─────────────────────────────────┘
```

### CODE REVIEW

File: crates/polars-plan/src/plans/optimizer/predicate_pushdown/mod.rs [hunk 1]
```diff
@@ -16,19 +16,19 @@ use crate::prelude::optimizer::predicate_pushdown::join::process_join;
 use crate::prelude::optimizer::predicate_pushdown::rename::process_rename;
 use crate::utils::{check_input_node, has_aexpr};
 
-pub type HiveEval<'a> =
+pub type ExprEval<'a> =
```
Comment (nameexhaustion) [inline crates/polars-plan/src/plans/optimizer/predicate_pushdown/mod.rs:19]: drive-by - rename to ExprEval as it is a generic evaluator

**Comment by ritchie46** [comment]
Nice one! Can you rebase?

**Comment by nameexhaustion** [comment]
rebased 👍
